python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -352,7 +352,7 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
             "max_task_retries" not in ray_remote_args
             and ray_remote_args.get("max_restarts") != 0
         ):
-            ray_remote_args["max_task_retries"] = 5
+            ray_remote_args["max_task_retries"] = -1
```

Comment:
Defaults are specified all over the code base. We should really consolidate them at some point. Maybe something like https://github.com/ray-project/ray/pull/39797.

I took a stab at it, but it's a bit more involved than I expected, so I'm leaving it to a future PR.